Source/Core/VideoCommon/VertexShaderManager.cpp

```diff
@@ -168,8 +168,14 @@ bool VertexShaderManager::UseVertexDepthRange()
     return true;
 
   // If an inverted depth range is unsupported, we also need to check if the range is inverted.
-  if (!g_ActiveConfig.backend_info.bSupportsReversedDepthRange && xfmem.viewport.zRange < 0.0f)
-    return true;
+  if (!g_ActiveConfig.backend_info.bSupportsReversedDepthRange)
```

Comment:
This isn't my strong suit but looking at this a while I wonder if it'd be slightly more informative to match the original source you referenced.

```
if (g_ActiveConfig.backend_info.bSupportsReversedDepthRange)
{
  return fabs(xfmem.viewport.zRange) > 16777215.0f || fabs(xfmem.viewport.farZ) > 16777215.0f;
}
else
{
  return xfmem.viewport.zRange < 0.0f || xfmem.viewport.zRange > 16777215.0f ||
            fabs(xfmem.viewport.farZ) > 16777215.0f;
}
```

This brings some symmetry to the supports reversed depth range vs not and gives a clearer picture of the requirements for each.

I'm assuming that this would also fix the issue you've reported based on your comment.  If I'm wrong, please let me know!

Reply:
I don't really have a source, I just followed the logic in BPFunctions.cpp which is tied to this.

I've tested your code revision and it no longer resolves the issue because the newly added xfmem.viewport.zRange > xfmem.viewport.farZ check is not present

Reply:
I was hesitant about your `zRange > farZ` check because it seems like that could trigger in situations when we don't want it to do so.  In particular, the bisect you referenced did not have that check (unless I missed it).  Which is why I wanted you to test the code I posted which more directly references the previous logic.

But if it doesn't resolve the issue...I will have to look at this in a debugger.  Not sure when I'll get to it.

Your check might be correct, I just don't feel confident in my understanding of z buffer details to really say for sure.

Thank you for testing, and for this PR!

Reply:
Ah, looking again.  I think I see it now.  You are trying to handle the case where `farZ - zRange` is negative.  Previously, in that bisect, the logic was clamping.  But not in the `VertexShaderManager` where I thought the logic was being done but rather in d3d's `SetViewport` function.  Hmm, your solution might be appropriate then.  Let me give that a bit more thought.